threestudio/systems/imagedreamfusion.py

```diff
@@ -109,63 +119,109 @@ def training_step(self, batch, batch_idx):
+            self.guidance.set_min_max_steps(
+                self.C(self.guidance.cfg.min_step_percent),
+                self.C(self.guidance.cfg.max_step_percent),
+            )
             prompt_utils = self.prompt_processor()
             guidance_out = self.guidance(
-                out["comp_rgb"], prompt_utils, **batch, rgb_as_latents=False
+                out["comp_rgb"],
+                prompt_utils,
+                **batch,
+                rgb_as_latents=False,
+                guidance_eval=guidance_eval,
             )
-
-        loss = 0.0
-        for name, value in guidance_out.items():
-            self.log(f"train/{name}", value)
-            if name.startswith("loss_"):
-                loss += value * self.C(self.cfg.loss[name.replace("loss_", "lambda_")])
+            if guidance_eval:
+                guidance_out, guidance_eval_out = guidance_out
+            set_loss("sds", guidance_out["loss_sds"])
 
         if self.C(self.cfg.loss.lambda_orient) > 0:
             if "normal" not in out:
                 raise ValueError(
                     "Normal is required for orientation loss, no normal is found in the output."
                 )
-            loss_orient = (
-                out["weights"].detach()
-                * dot(out["normal"], out["t_dirs"]).clamp_min(0.0) ** 2
-            ).sum() / (out["opacity"] > 0).sum()
-            self.log("train/loss_orient", loss_orient)
-            loss += loss_orient * self.C(self.cfg.loss.lambda_orient)
+            set_loss(
+                "orient",
+                (
+                    out["weights"].detach()
+                    * dot(out["normal"], out["t_dirs"]).clamp_min(0.0) ** 2
+                ).sum()
+                / (out["opacity"] > 0).sum(),
+            )
 
         if self.C(self.cfg.loss.lambda_normal_smooth) > 0:
+            if "comp_normal" not in out:
+                raise ValueError(
+                    "comp_normal is required for 2D normal smooth loss, no comp_normal is found in the output."
+                )
+            normal = out["comp_normal"]
+            set_loss(
+                "normal_smooth",
+                (normal[:, 1:, :, :] - normal[:, :-1, :, :]).square().mean()
+                + (normal[:, :, 1:, :] - normal[:, :, :-1, :]).square().mean(),
+            )
+
+        if self.C(self.cfg.loss.lambda_3d_normal_smooth) > 0:
             if "normal" not in out:
                 raise ValueError(
                     "Normal is required for normal smooth loss, no normal is found in the output."
                 )
-            normal = out["normal"]
-            loss_normal_smooth = (
-                normal[:, 1:, :, :] - normal[:, :-1, :, :]
-            ).square().mean() + (
-                normal[:, :, 1:, :] - normal[:, :, :-1, :]
-            ).square().mean()
-            self.log("train/loss_normal_smooth", loss_normal_smooth)
-            loss += self.C(self.cfg.loss.lambda_normal_smooth) * loss_normal_smooth
```

Comment:
Was this removed on purpose? I recommend keeping this loss as well as 3d_normal_smooth in case `normal_smooth` is much cheaper computationally (less rays) and still provides a good regularization.

Reply:
No it's present in the lines above 160-165. Just a mismatch in tracking changes